lib/shared/types/src/types/config/models/variable.ts

```diff
@@ -41,6 +41,7 @@ export enum VariableType {
  */
 export type DVCJSON = { [key: string]: string | boolean | number }
 export type VariableValue = string | boolean | number | DVCJSON
+export type ArrayVariableValue = VariableValue | Array<string | boolean | number>
```

Comment:
arrays can contain any valid JSON so we should make sure DVCJSON is in here

Reply:
actually one thing I just realized is that the VariableValue type definition shouldn't even include DVCJSON, since we don't restrict variable values to one level of nesting. I think the correct type there would just be a `Record<string, unknown>` or something